docs/source/guide/scripts.md

```diff
@@ -0,0 +1,211 @@
+console.log("Plotly is ready");
+```
+
+##### `on(eventName, handler)`
+
+Subscription to listen to events related to the Label Studio Frontend. All handlers added within the framework of this method will be unsubscribed when switching to another annotation.
+
+For a list of all available events, see our [Frontend reference](frontend_reference#Available-events).
+
+| Parameter &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; | Type     | Description          |
+|--------------------------|--------------------|--------------------------|
+| `eventName` | string | A case-sensitive string representing the event type to listen for.                  |
+| `handler`  | function | Allows a browser to check the fetched script to ensure that the code is never loaded if the source has been manipulated.  |
+
+
+##### `get dataObj`
+
+Alias to `.task.data`. This is the core data structure for a task, and includes the original data that needs to be annotated.
+
+##### `get task`
+
+A getter that returns information about current task:
+* `id` - ID of the task. 
+* `data` - Object representing task data. 
+
+##### `get annotation`
+
+A getter that returns the currently selected annotation.
+
+##### `get regions`
+
+A getter that returns all regions of the current annotation.
+
+
+## Frontend API implementation details
+
+The following implementation details may be useful when creating your own custom scripts. 
+
+!!! note
+    While these details are relatively stable, we make no guarantees that they will not change in the future. 
+
+
+### Regions
+
+!!! note
+    These are used within `Htx.annotationStore.selected`. For example, `.regions` would be specified as `Htx.annotationStore.selected.regions`. For an example of this in use, see [HyperText video and time series sync](script_examples#HyperText-video-and-time-series-sync), [Bulk text labeling with regex](script_examples#Bulk-text-labeling-with-regex), and [Bulk creation and deletion operations with keyboard shortcut](script_examples#bulk-creation-and-deletion-operations-with-keyboard-shortcut). 
+
+
+For more information on how regions are stored and formatted, see [How Label Studio saves results in annotations](task_format#How-Label-Studio-saves-results-in-annotations). 
+
+##### `.areas` 
+
+Map of regions.
+
+##### `.regions` 
+
+An array of all regions (includes classifications). 
+
+##### `.regionStore.regions` 
+
+An array of all real regions (excludes classifications).
+
+##### `.results` 
+
+Array of all results. 
+
+Note that this returns a large object with keys of all possible result types, but only one result type has an actual value. The value of this key is populated by the `valueType` of the related control tag, which is typically the same as the `type` for the result. To access this value directly, use `result.mainValue`. 
```

Comment:
I'd leave only the part about `result.mainValue`, the rest is very confusing. Maybe to shortly mention that `result.mainValue` is a shortcut for `r[control.valueType]` and the other keys are useless.

And "this returns a large object" is factually incorrect, the previous line says that this is an array of objects.

Reply:
I updated this to remove the second sentence, but left the first sentence because @jombooth had told me the bit about only result type having an actual value is important.